alica_engine/include/engine/Logging.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Logging will be revised with S.11 of the ALICA Framework Roadmap.
+ */
+
+#pragma once
+
+#include <iostream>
+#include <string>
+#include "engine/AlicaEngine.h"
+
+namespace alica
+{
+namespace logging
+{
+std::string getLogFilename(AlicaEngine* ae, const std::string& file);
+std::string getErrLogFilename(AlicaEngine* ae, const std::string& file);
+
+struct None
+{
+};
+
+template <typename List>
+struct LogData
+{
+    List list;
+};
+
+template <typename Begin, typename Value>
+constexpr LogData<std::pair<Begin&&, Value&&>> operator<<(LogData<Begin>&& begin, Value&& value) noexcept
+{
+    return {{std::forward<Begin>(begin.list), std::forward<Value>(value)}};
+}
+
+template <typename Begin, size_t n>
+constexpr LogData<std::pair<Begin&&, const char*>> operator<<(LogData<Begin>&& begin, const char (&value)[n]) noexcept
+{
+    return {{std::forward<Begin>(begin.list), value}};
+}
+
+typedef std::ostream& (*PfnManipulator)(std::ostream&);
+
+template <typename Begin>
+constexpr LogData<std::pair<Begin&&, PfnManipulator>> operator<<(LogData<Begin>&& begin, PfnManipulator value) noexcept
+{
+    return {{std::forward<Begin>(begin.list), value}};
+}
+
+template <typename Begin, typename Last>
+void output(std::ostream& os, std::pair<Begin, Last>&& data)
+{
+    output(os, std::move(data.first));
+    os << data.second;
+}
+
+inline void output(std::ostream& os, None) {}
+
+#define NOINLINE_ATTRIBUTE
+
+template <typename List>
+void Log(const char* file, const char* function, int line, LogData<List>&& data) NOINLINE_ATTRIBUTE
+{
+    std::cout << "[" << file << "::" << function << ":" << line << "] ";
+    output(std::cout, std::move(data.list));
+    std::cout << std::endl;
+}
+
+#define LOG(msg) (alica::logging::Log(basename(__FILE__), __FUNCTION__, __LINE__, alica::logging::LogData<alica::logging::None>() << msg))
```

Comment:
Loggin was part of system_util. system_util is a dependency of alica_engine, but Loggin is using the old SystemConfig. Because system_util can not depend on alica_engine, where the new config is placed, I talked with Stephan and we decided to move Logging into alica_engine for now, as logging will be revised with S.11 of the ALICA Framework Roadmap. All Tests do run even without the Logging class. If no other features from other repos are using this class, this can also be removed entirely.